ambry-cloud/src/main/java/com/github/ambry/cloud/RecoveryNetworkClient.java

```diff
@@ -67,17 +70,38 @@
 public class RecoveryNetworkClient implements NetworkClient {
   private final static Logger logger = LoggerFactory.getLogger(RecoveryNetworkClient.class);
   private final ClusterMap clustermap;
-  private final FindTokenHelper findTokenHelper;
   private final StoreManager storeManager;
-  private final ConcurrentHashMap<StoreKey, MessageInfo> messageInfoCache = new ConcurrentHashMap<>();
-  protected final CosmosContainer cosmosContainer;
+  private final ConcurrentHashMap<StoreKey, MessageInfo> messageInfoCache;
+  private final AzureBlobLayoutStrategy azureBlobLayoutStrategy;
+  private final ClusterMapConfig clusterMapConfig;
+  private final AzureCloudConfig azureCloudConfig;
+  private final AzureCloudDestinationSync azureSyncClient;
+  private final RecoveryNetworkClientCallback clientCallback;
+  private final RecoveryMetrics recoveryMetrics;
 
-  public RecoveryNetworkClient(ClusterMap clustermap, FindTokenHelper findTokenHelper, StoreManager storeManager,
-      CosmosContainer cosmosContainer) {
-    this.clustermap = clustermap;
-    this.findTokenHelper = findTokenHelper;
+  class EmptyCallback implements RecoveryNetworkClientCallback {
+    public void onListBlobs(ReplicaMetadataRequestInfo request) {}
+  }
+
+  public RecoveryNetworkClient(VerifiableProperties properties, MetricRegistry registry, ClusterMap clusterMap,
+      StoreManager storeManager, AccountService accountService, RecoveryNetworkClientCallback clientCallback) {
+    this.clientCallback = clientCallback == null ? new EmptyCallback() : clientCallback;
+    this.clustermap = clusterMap;
     this.storeManager = storeManager;
-    this.cosmosContainer = cosmosContainer;
+    this.clusterMapConfig = new ClusterMapConfig(properties);
+    this.azureCloudConfig = new AzureCloudConfig(properties);
+    this.recoveryMetrics = new RecoveryMetrics(registry);
+    this.messageInfoCache = new ConcurrentHashMap<>();
```

Comment:
Looks like we don't use the cache for this version. 
Do we use cache before for the restore? How did we use it?

Reply:
Brought it back, the cache is used to store metadata fetched during the first part of replication - get-metadata. This  is later used to fetch blob data. This reduces the number of metadata calls to azure.